### ISSUE TITLE
json unmarshall error on SnatRule.source_type

### ISSUE THREAD

**Issue by kzap** [huaweicloud/terraform-provider-huaweicloud#297]
> Error: Error creatting Snat Rule: json: cannot unmarshal number into Go struct field SnatRule.source_type of type string

I created a SNAT rule and I got this error but the SNAT Rule was successfully created:

### Terraform Version
`0.12.24`

### Affected Resource(s)
- huaweicloud_nat_snat_rule_v2

### Terraform Configuration Files
```hcl
resource "huaweicloud_nat_snat_rule_v2" "nat_az1_snat" {
  nat_gateway_id = huaweicloud_nat_gateway_v2.nat_az1.id
  network_id     = local.network_uuid1
  floating_ip_id = huaweicloud_networking_floatingip_v2.floatip_nat_az1.id
}
```

### Expected Behavior
SNAT rule added to state

### Actual Behavior
SNAT rule was created and not added to state so Terraform keeps trying to recreate it.

### Steps to Reproduce
1. `terraform apply`

**Comment by niuzhenguo** [comment]
@kzap This already fixed by #291 please try to build with the master branch.

**Comment by niuzhenguo** [comment]
@kzap closing this, please feel free to reopen it if the problem still exist in v1.14.0.
